apps/frontend/src/app/pages/maps/map-edit/map-edit.component.ts

```diff
@@ -184,24 +185,27 @@ export class MapEditComponent implements OnInit, ConfirmDeactivate {
     })
   });
 
-  versionForm = this.fb.group({
-    bsp: [
-      null,
-      [Validators.required, FileValidators.maxSize(MAX_BSP_SIZE)],
-      [FileValidators.isCompressedBsp()]
-    ],
-    zon: [null, [], [FileValidators.isValidZones()]],
-    vmfs: [
-      [],
-      [FileValidators.maxSize(MAX_VMF_SIZE), FileValidators.extension('vmf')],
-      [FileValidators.isValidVdf()]
-    ],
-    changelog: [
-      '',
-      [Validators.required, Validators.maxLength(MAX_CHANGELOG_LENGTH)]
-    ],
-    resetLbs: [false]
-  });
+  versionForm = this.fb.group(
+    {
+      bsp: [
+        null,
+        [FileValidators.maxSize(MAX_BSP_SIZE)],
+        [FileValidators.isCompressedBsp()]
+      ],
+      zon: [null, [], [FileValidators.isValidZones()]],
+      vmfs: [
+        [],
+        [FileValidators.maxSize(MAX_VMF_SIZE), FileValidators.extension('vmf')],
+        [FileValidators.isValidVdf()]
+      ],
+      changelog: [
+        '',
+        [Validators.required, Validators.maxLength(MAX_CHANGELOG_LENGTH)]
+      ],
+      resetLbs: [false]
+    },
+    { validators: atLeastOneFileRequired(['bsp', 'zon']) }
```

Comment:
Should allow for VMFs as well, could imagine someone only uploading a BSP and not VMF, then deciding to the VMF after.